### PR TITLE
Django upgrade

### PR DESCRIPTION
**Pull request by abztrakt**
candidate for v0.4

### CODE REVIEW

File: uw_spotseeker/__init__.py [hunk 1]
```diff
@@ -144,102 +108,100 @@ def search_spots(self, query_tuple):
...
-            raise DataFailureException(url, response.status, response.data)
-        results = json.loads(response.data.decode('utf-8'))
+        if status != 200:
+            raise DataFailureException(url, status, content)
+        results = json.loads(content.decode('utf-8'))
 
         return self._spots_from_data(results)
 
-    def put_spot(self, spot_id, spot_json, etag):
+    def put_spot(self, spot_id, spot_json: str, etag) -> Tuple[MockHTTP, dict]:
```
Comment (abztrakt) [inline uw_spotseeker/__init__.py:120]: Do we think this type hint is correct?

File: docker/test-app/test_app/views.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+from uw_spotseeker import Spotseeker
+import json
+import logging
+
+logging.basicConfig()
+logger = logging.getLogger(__name__)
+
+
+def get_token() -> str:
+    token = cache.get(settings.APP_NAME)
```
Comment (David-Strupinski) [inline docker/test-app/test_app/views.py:18]: Can be simplified into one line:
`return cache.get(settings.APP_NAME, 'no token in cache')`

File: docker/test-app/test_app/templates/all_spots.html [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+      text-decoration: none;
+      color: inherit;
+  }
+</style>
+{% endblock %}
+
+
+{% block scripts %}
+
+<script>
```
Comment (David-Strupinski) [inline docker/test-app/test_app/templates/all_spots.html:85]: move styles and scripts to separate files or keep inline?

File: docker/test-app/test_app/templates/base.html [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+<html>
+<head>
+    <title>{{ title }}</title>
+</head>
+<body>
+    <h1>{{ title }}</h1>
+
+    <div><b>Scope:</b> {{ scope }}</div>
+    <div><b>Access Token:</b> {{ token }}</div>
+    <br />
```
Comment (David-Strupinski) [inline docker/test-app/test_app/templates/base.html:11]: `<br>`

File: docker/test-app/test_app/templates/spot.html [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+
+<p>{{ spot.latitude }} {{ spot.longitude }}</p>
+
+<p>{{ spot.building_name }}</p>
+
+<p>{{ spot.capacity }}</p>
+
+<p>{{ spot.last_modified }}</p>
+
+<b>Extended Info:</b>
```
Comment (David-Strupinski) [inline docker/test-app/test_app/templates/spot.html:37]: replace all `<b>` with `<strong>`

File: docker/test-app/test_app/templates/spot.html [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+        // get image id from form
+        const imageId = event.target.dataset.imageid;
+        const isItem = event.target.dataset.item;
+        var imageUrl;
+
+        if (isItem) {
+            var itemId = event.target.dataset.itemid;
+            imageUrl = `/item/${itemId}/image/${imageId}`;
+        } else {
+            imageUrl = `${document.URL}/image/${imageId}`;
```
Comment (David-Strupinski) [inline docker/test-app/test_app/templates/spot.html:110]: replace `${document.URL}` with /spot/{{ spot.spot_id  }} ?

File: docker/test-app/test_app/templates/spot_edit.html [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+    </div>
+
+    <div class="form-group">
+        <label for="latitude">Latitude</label>
+        <input type="number" id="latitude" name="latitude" value="{{ spot.latitude }}" class="form-control">
+    </div>
+
+    <div class="form-group">
+        <label for="longitude">Longitude</label>
+        <input type="number" id="longitude" name="longitude" value="{{ spot.longitude }}" class="form-control">
```
Comment (David-Strupinski) [inline docker/test-app/test_app/templates/spot_edit.html:20]: these ids aren't necessary

**Pull request merged**